control/controlhttp/constants.go

```diff
@@ -28,33 +31,43 @@ const (
 	serverUpgradePath = "/ts2021"
 )
 
-// DialOpts contains options when calling Dial.
-type DialOpts struct {
-	// The following fields are required
-
+// Dialer contains configuration on how to dial the Tailscale control server.
+type Dialer struct {
```

Comment:
can the controlhttp move to a Dialer type be in its own change first without any dialplan stuff?

Reply:
Sure, lemme break this out into a new PR

Reply:
#5661